internal/pkg/cli/task_run.go

```diff
@@ -481,6 +504,70 @@ func (o *runTaskOpts) Execute() error {
 	return nil
 }
 
+func (o *runTaskOpts) generateCommand() error {
+	g, err := o.configureGenerator()
+	if err != nil {
+		return err
+	}
+	command, err := g.Generate()
+	if err != nil {
+		return fmt.Errorf("generate command: %w", err)
+	}
+	log.Infoln(command.String())
+	return nil
+}
+
+func (o *runTaskOpts) configureGenerator() (cmdGenerator, error) {
+	var g cmdGenerator
+	sess, err := sessions.NewProvider().Default()
+	if err != nil {
+		return nil, fmt.Errorf("get default session: %s", err)
+	}
+
+	if arn.IsARN(o.generateCommandTarget) {
+		return o.configureGeneratorFromARN(sess)
+	}
+
+	parts := strings.Split(o.generateCommandTarget, "/")
+	switch len(parts) {
+	case 2:
+		clusterName, serviceName := parts[0], parts[1]
+		g = generator.ECSServiceCommandGenerator{
+			Cluster:   clusterName,
+			Service:   serviceName,
+			ECSClient: awsecs.New(sess),
+		}
+	case 3:
+		appName, envName, serviceName := parts[0], parts[1], parts[2]
+		g = generator.ServiceCommandGenerator{
+			App:                  appName,
+			Env:                  envName,
+			Service:              serviceName,
+			ECSInformationGetter: ecs.New(sess),
+		}
+	default:
+		return nil, errors.New("invalid input to --generate-cmd")
+	}
+	return g, nil
+}
+
+func (o *runTaskOpts) configureGeneratorFromARN(sess *session.Session) (cmdGenerator, error) {
+	svcARN := awsecs.ServiceArn(o.generateCommandTarget)
+	clusterName, err := svcARN.ClusterName()
+	if err != nil {
+		return nil, fmt.Errorf("extract cluster name from arn %s", svcARN)
+	}
+	serviceName, err := svcARN.ServiceName()
+	if err != nil {
+		return nil, fmt.Errorf("extract service name from arn %s", svcARN)
+	}
+	return generator.ECSServiceCommandGenerator{
```

Comment:
Are we not supporting jobs with this feature? I apologize I didn't get in on the earlier review of this. Or was the decision made to delegate that to the hypothetical `job run` subcommand?

Reply:
I will make a draft PR for job for this feature! The code is working but is a little messy and I haven't got the chance to really clean it up, so it's not in this PR yet.